File: Backend/entity/employee.go
package entity

import "gorm.io/gorm"

type Employee struct {
	gorm.Model
	FirstName         string `valid:"required~FirstName is required"`
	LastName          string `valid:"required~LastName is required"`
	EMPCode        	  string `gorm:"uniqueIndex;not null" valid:"required~EMPCode is required"`
	PhoneNumber		  string `valid:"required~PhoneNumber is required"`
	Email             string `valid:"required~Email is required, email~Email is invalid"`
	Profile			  string `gorm:"type:text" valid:"required~Profile is required"`
	Password          string `valid:"required~Password is required"`
	BankAccountNumber string `valid:"required~BankAccountNumber is required"`
	Line              string `valid:"required~Line is required"`
	
	BankTypeID        uint		`valid:"required~BankTypeID is required"`
	BankType          BankType 	`gorm:"foreignKey:BankTypeID"`
	
	RoleID            uint	`valid:"required~RoleID is required"`
	Role              Role 	`gorm:"foreignKey:RoleID"`

	Cart      []Cart      `gorm:"foreignKey:EmployeeID"`
	OrderBill []OrderBill `gorm:"foreignKey:EmployeeID"`
	Bill      []Bill      `gorm:"foreignKey:EmployeeID"`
}

File: Frontend/src/interfaces/Employee.ts
export interface EmployeeInterface {
    ID?: number;
    FirstName?: string;
    LastName?: string;
    PhoneNumber?: string;
    EMPCode? : string;
    Email?: string;
    Password?: string;
    Profile?: string;
    RoleID?: number;
    BankAccountNumber?: number;
    Line?: string;

    Role? : {
        ID?: number;
        RoleName?: string;
    }

    BankType? : {
        ID?: number;
        BankTypeName?: string;
    }
}